src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -505,13 +488,29 @@ internal sealed class AnalyzerInconsistencyBuildResponse : BuildResponse
 
     internal sealed class RejectedBuildResponse : BuildResponse
     {
+        public string Reason;
+
         public override ResponseType Type => ResponseType.Rejected;
 
+        public RejectedBuildResponse(string reason)
+        {
+            Reason = reason;
+        }
+
         /// <summary>
         /// AnalyzerInconsistency has no body.
         /// </summary>
         /// <param name="writer"></param>
-        protected override void AddResponseBody(BinaryWriter writer) { }
+        protected override void AddResponseBody(BinaryWriter writer)
+        {
+            WriteLengthPrefixedString(writer, Reason);
+        }
+
+        public static RejectedBuildResponse Create(BinaryReader reader)
+        {
+            var reason = ReadLengthPrefixedString(reader) ?? "";
```

Comment:
Do we want to put something other than "" here for the reason? Presumably something very weird is going on if ReadLengthPrefixedString() is returning null.